biloba/config.py

```diff
@@ -99,6 +101,25 @@ def get_key(config, key, default=missing):
     return val
 
 
+def set_value(config, key, value):
+    """
+    Uses a dotted notation to set a value in a dict.
+    """
+    try:
+        path, key = key.rsplit('.', 1)
+    except ValueError:
+        return config.__setitem__(key, value)
+
+    obj = get_key(config, path)
+
+    if not isinstance(obj, dict):
```

Comment:
I would suggest just trying the operation (duck typing) so that non dict objects can be supported (implement ``__setitem__``).